packages/core/src/lib/session.ts

```diff
@@ -41,6 +42,30 @@ export const assignInteractionResults = async (
   ctx.body = { redirectTo, ts };
 };
 
+export const checkSessionHealth = async (
+  ctx: Context,
+  provider: Provider,
+  tolerance = 10 * 60 // 10 mins
+) => {
+  const { result } = await provider.interactionDetails(ctx.req, ctx.res);
+
+  if (!result?.login?.accountId) {
+    throw new RequestError('auth.unauthorized');
+  }
+
+  if (!result.login.ts || result.login.ts < dayjs().unix() - tolerance) {
+    const { passwordEncrypted, primaryPhone, primaryEmail } = await findUserById(
+      result.login.accountId
+    );
+
+    // No authenticated method configured for this user. Pass!
+    if (!passwordEncrypted && !primaryPhone && !primaryEmail) {
+      return;
+    }
+    throw new RequestError('auth.require_re_authentication');
```

Comment:
If result.login.ts does not exists, should we throw specific error?

Reply:
I don't think that's necessary, if there's no `ts`, the only possible reason I can think of right now is the user hasn't logged in.